src/components/Navigation/TopBar/Notifications/Notifications.tsx

```diff
@@ -0,0 +1,230 @@
+  border-bottom: 1px solid ${colors.ui.background__medium.hex};
+`;
+
+export const UnreadRedDot = styled.div`
+  background-color: ${colors.interactive.danger__hover.rgba};
+  width: 12px;
+  height: 12px;
+  border-radius: 50%;
+  position: absolute;
+  right: 7px;
+  top: 7px;
+  box-shadow:
+    0 2px 4px rgba(0, 0, 0, 0.14),
+    0 3px 4px rgba(0, 0, 0, 0.12);
+  border: 1.5px solid ${colors.text.static_icons__primary_white.hex};
+`;
+
+const FilterOptionsContainer = styled.div`
+  display: flex;
+`;
+
+interface NotificationsProps {
+  setAllAsRead: () => void;
+  hasUnread?: boolean;
+  showFilterOptions?: boolean;
+  children?: ReactNode;
+  notifications?: (
+    | ReadyToReportNotificationTypes
+    | RequestChangeOrcaTypes
+    | MergeBranchOrcaTypes
+    | Due3WeeksTypes
+    | ExperienceReadyToPublishTypes
+    | ReviewQANotificationsTypes
+    | DefaultNotificationTypes
+    | RequestReviewOrcaTypes
+  )[];
+}
+
+export const Notifications: FC<NotificationsProps> = ({
+  children,
+  hasUnread = false,
+  setAllAsRead,
+  showFilterOptions = false,
+  notifications,
+}) => {
+  const sidePanelRef = useRef<HTMLDivElement | null>(null);
+  const buttonRef = useRef<HTMLDivElement | null>(null);
+  const filterMenuRef = useRef<HTMLDivElement | null>(null);
+  const sortMenuRef = useRef<HTMLDivElement | null>(null);
+
+  const [notificationsOpen, setNotificationsOpen] = useState(false);
+
+  const [filteringOn, setFilteringOn] = useState<FilterNotification[]>([]);
+
+  const [sortingOn, setSortingOn] = useState<SortNotification[]>([]);
+
+  const filteredAndSortedNotifications = useMemo(() => {
+    if (!notifications) return [];
+    let copy = [...notifications];
+
+    if (filteringOn.length > 0) {
+      copy = copy.filter((notification) => {
+        if (filteringOn.includes(FilterNotification.UNREAD)) {
+          return !notification.Read;
+        } else if (filteringOn.includes(FilterNotification.USER)) {
+          return notification.user;
+        } else if (filteringOn.includes(FilterNotification.SYSTEM)) {
+          return !notification.user;
+        }
+      });
+    }
+    if (sortingOn.length > 0) {
+      copy = copy.sort((a, b) => {
+        const aUnixTime = new Date(a.time).getTime();
+        const bUnixTime = new Date(b.time).getTime();
+
+        if (sortingOn.includes(SortNotification.OLD_NEWEST)) {
+          return bUnixTime - aUnixTime;
+        } else if (sortingOn.includes(SortNotification.UNREAD)) {
+          return a.Read === b.Read ? 0 : a.Read ? 1 : -1;
+        } else {
+          return aUnixTime - bUnixTime;
+        }
+      });
+    }
+
+    return copy;
+  }, [filteringOn, notifications, sortingOn]);
+
+  const handleButtonClick = () => {
+    if (notificationsOpen) {
+      onClose();
+    } else {
+      setNotificationsOpen(true);
+    }
+  };
+
+  const onClose = () => {
+    setAllAsRead();
+    setNotificationsOpen(false);
+  };
+
+  useOutsideClick(sidePanelRef.current, (event) => {
```

Comment:
Use onClose on the menu instead of this 'useOutsideClick' hook

Reply:
but the sidepanel is a div and not a menu, can I still have the onClose on that ?